### PR TITLE
Preventing unconnected nodes or subgrids.

### LINKED ISSUE #415: Identify unconnected nodes or subgrids and treat them appropriately
grafik

### PR DESCRIPTION
**Pull request by staudtMarius**
Resolves #415

### CODE REVIEW

**Comment by staudtMarius** [comment]
!test

File: src/main/scala/utils/Clustering.scala [hunk 1]
```diff
@@ -229,16 +299,20 @@ object Clustering {
     * @return
     *   either the total distance or [[Double.MaxValue]]
     */
-  private def totalDistance(list: List[Cluster]): Double = {
-    list.map { l => l.distances }.reduceOption { (a, b) => a + b } match {
-      case Some(value) => value
-      case None        => Double.MaxValue
+  private def totalDistance(list: Set[Cluster]): Double = {
```
Comment (danielfeismann) [inline src/main/scala/utils/Clustering.scala:302]: Please try if you find a naming that comply with sonar although it was named like this before.

File: src/main/scala/utils/Clustering.scala [hunk 2]
```diff
@@ -6,125 +6,193 @@
...
   /** Method to run the algorithm. This algorithm is based on PAM (Partitioning
     * Around Medoids).
+    *
     * @param maxIteration
     *   maximal number of iterations
     * @return
     *   a list of [[Cluster]]s
     */
-  def run(implicit maxIteration: Int = 10): List[Cluster] = {
+  def run(implicit maxIteration: Int = 50): List[Cluster] = {
```
Comment (danielfeismann) [inline src/main/scala/utils/Clustering.scala:42]: What are the reasons to increase the number of maxIterations?

**Review by danielfeismann — COMMENTED** [review]
Need to have a deeper look here later

File: src/main/scala/edu/ie3/osmogrid/lv/LvGridGeneratorSupport.scala [hunk 3]
```diff
@@ -266,166 +256,50 @@ object LvGridGeneratorSupport extends LazyLogging {
     }
   }
 
-  /** Recursively traverses the graph by starting at a given node and follows
-    * all connected edges sequentially, building all lines between nodes in the
-    * process.
+  /** This method will reduce the graph by removing some vertices. A vertex is
+    * removed if its degree is < 2 and it is not defined to be kept.
```
Comment (danielfeismann) [inline src/main/scala/edu/ie3/osmogrid/lv/LvGridGeneratorSupport.scala:260]:
```suggestion
    * removed if its degree is <= 2 and it is not defined to be kept.
```

**Review by danielfeismann — CHANGES_REQUESTED** [review]
Beside of two minor things this looks good to me

**Review by danielfeismann — APPROVED** [review]
LGTM!

**Pull request merged**